### ISSUE TITLE
Erreur de génération du lien Iframe (dans l'option "partager")

### ISSUE THREAD

**Issue by Clement-Bader-bzh** [mviewer/mviewer#955]
## Environnement

Version mviewer: 3.9.1

Navigateur: Chrome + Firefox

## Description

Lorsqu'il y a génération d'un permalien, le code d'Iframe proposé n'est pas nécessairement le bon car il conserve les couches par défaut de l'application (même si elles ont été retirées pour générer l'Iframe). 

## Ce que je devrai voir

Le code d'Iframe devrait intégrer l'Url du permalien (alors qu'ici il intégère également les couches par défaut).

## Etapes

- Utiliser une application avec des couches affichées à l'ouverture.
- Les retirer (et éventuellement en afficher d'autres)
- Utiliser l'option "partager cette carte"
- Comparer le permalien généré et l'URL intégrée dans de code d'Iframe proposée (option "src=")

**Comment by Gaetanbrl** [comment]
Bonjour et merci pour cette issue,
Si j'ai bien compris, il est attendu d'avoir le permalien dans l'URL d'iframe car vous observer une différence c'est bien cela ?

**Comment by Clement-Bader-bzh** [comment]
> Bonjour et merci pour cette issue, Si j'ai bien compris, il est attendu d'avoir le permalien dans l'URL d'iframe car vous observer une différence c'est bien cela ?

Oui, tout à fait (pour permettra l'affichage à l'identique du permalien au sein de l'iframe)

**Comment by lecault** [comment]
@Clement-Bader-bzh peux tu me dire sur quelle apps tu as vu cela ? Je ne reproduit pas...

**Comment by Gaetanbrl** [comment]
Le problème est reproductible avec l'outil de partage :

1. aller sur la démo simple : https://kartenn.region-bretagne.fr/kartoviz/?config=demo/geobretagne.xml#
2. décocher la couche `PDIPR 35`
3. cliquer sur "partager"
4. cliquer sur "Lien permanent vers la carte"
5. Voir que la couche est visible alors qu'on l'a décoché

https://kartenn.region-bretagne.fr/kartoviz/?x=-198478&y=6137005&z=14&l=&lb=photo_actuelle&config=demo/geobretagne.xml&mode=d#

6. refaire les étapes sans faire l'étape 2

7. Voir que le lien inclu bien la couche mais qu'on a aucune différence avec le résultat du lien précédent

https://kartenn.region-bretagne.fr/kartoviz/?x=-198478&y=6137005&z=14&l=pdipr_35*cd35%3APDIPR35-Pedestre&lb=photo_actuelle&config=demo/geobretagne.xml&mode=d

**Comment by Gaetanbrl** [comment]
Cette issue est potentielelment liée à https://github.com/mviewer/mviewer/pull/860

**Comment by Agath21** [comment]
Je rebondis, après test décrit dans l'issue #967, il semble y avoir en effet un problème avec l'url de partage quand le paramètre `l=` est vide (aucune couche n'est séléctionnée sur la carte lors du partage) : 
`https://kartenn.region-bretagne.fr/kartoviz/?x=-198478&y=6137005&z=14&l=&lb=photo_actuelle&config=demo/geobretagne.xml&mode=d`

Le mviewer affiche les couches par défaut.
